### ISSUE TITLE
should route to backend according to the key if server_connections > 1

### ISSUE THREAD

**Issue by idning** [twitter/twemproxy#353]
in the case of ``server_connections > 1``

if one proxy receive 2 pipelined request:

    set k v
    del k

it has possibility that the ``del`` and ``set`` cmd will go to different server_connection.

and ``del`` may executed before ``set``, so after this pipeline returned, we may find that the command in the pipeline is out of order. 

this will be a problem especially in the case of import lot's of data using pipeline.

**Comment by charsyam** [comment]
@idning yes, and actually twemproxy don't recommend server_connections > 1 in Redis.

**Comment by manjuraj** [comment]
@idning -- https://github.com/twitter/twemproxy/blob/master/notes/recommendation.md#server_connections--1
